src/tls.c

```diff
@@ -12137,16 +12137,25 @@ int TLSX_PopulateExtensions(WOLFSSL* ssl, byte isServer)
             if (usingPSK)
             #endif
             {
-                byte modes;
+                byte modes = 0;
 
                 (void)usingPSK;
                 /* Pre-shared key modes: mandatory extension for resumption. */
-                modes = 1 << PSK_KE;
+            #ifdef HAVE_SUPPORTED_CURVES
+fprintf(stderr, "ONLY PSK DHE KE: %d\n", ssl->options.onlyPskDheKe);
+                if (!ssl->options.onlyPskDheKe)
+            #endif
+                {
+                    modes = 1 << PSK_KE;
+                }
             #if !defined(NO_DH) || defined(HAVE_ECC) || \
                               defined(HAVE_CURVE25519) || defined(HAVE_CURVE448)
-                if (!ssl->options.noPskDheKe)
+fprintf(stderr, "NO PSK DHE KE: %d\n", ssl->options.noPskDheKe);
+                if (!ssl->options.noPskDheKe) {
                     modes |= 1 << PSK_DHE_KE;
+                }
             #endif
+fprintf(stderr, "MODES: %d\n", modes);
```

Comment:
Did you mean for this to get included in the PR?

Reply:
Removed